### DISCUSSION TITLE
How to build RabbitMQ Client v3.6.9?

### DISCUSSION THREAD

**Discussion by ZedZipDev** [rabbitmq/rabbitmq-dotnet-client#1178] [General]
I have downloaded the RabbitMQ Client v3.6.9 and tried to build it in VS2019.
When I build it in VS2019 I receive:
error CS2001: Source file 'D:\MyProjects\RabbitMQ\rabbitmq-dotnet-client-rabbitmq_v3_6_9\projects\client\RabbitMQ.Client\..\..\..\gensrc\RabbitMQ.Client\autogenerated-api-0-9-1.cs' could not be found.

How to fix it?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by lukebakken** [comment]
@Oleg26Dev - please run `build.bat` from a VS command prompt first before attempting to build in Visual Studio. This will generate the necessary file.

**Comment by michaelklishin** [comment]
Why would you need a version from [five years ago](https://www.rabbitmq.com/changelog.html)? You can use the latest version of the client with RabbitMQ 3.6 (which has been out of support for years).

**Reply by ZedZipDev** [reply]
Hi , I understand that it is bad idea but what can I do?

Shortly what I do now. 
My goal is: to create SQL CLR functions for pub/receive messages to AWS RabbitMQ.

I have create a small NET Framework 4.8 app + RabbitMQ 6.* and tested connection to the RabbitMQ Docker - it works fine.
Also, it works fine with AWS RabbitMQ if I connect 
```
            ConnectionFactory connectionFactory = new ConnectionFactory
            {
                Uri = new Uri("amqps:/xyxy.amazonaws.com:5671") // amqps - is important
            };

            connectionFactory.UserName = "myuser";
            connectionFactory.Password = "mypwd5";
            try
            {
                IConnection connection = connectionFactory.CreateConnection();
                Console.WriteLine("OK");
            }
            catch (Exception x)
            {
                Console.WriteLine("***{0}", x.Message);
            }
```

If I use amqp it gives me an error: no endpoint ....

SQL Server 2017-19 supports NET 4.5 that's why I have corrected my app and use NET 4.5 + corresponding RabbitMQ Client (3.6.9)
RabbitMQ Client (3.6.9) does not use amqps but amqp only. And I receive again "no endpoint"

**Reply by ZedZipDev** [reply]
> Why would you need a version from [five years ago](https://www.rabbitmq.com/changelog.html)? You can use the latest version of the client with RabbitMQ 3.6 (which has been out of support for years).

Now I have reconfigured my test app: 

NET 4.5.1 + RabbitMQ Client 5.2.0. The code
ConnectionFactory connectionFactory = new ConnectionFactory
            {
                Uri = new Uri("amqps://abcd.amazonaws.com"),
                UserName = "admin",
                Password = "admin"
            };
and in the debug connectionFactory contains amqp(!)
Endpoint = {amqp://abcd.amazonaws.com:5671}
and logically: "No endpoint"

**Reply by lukebakken** [reply]
https://rabbitmq.github.io/rabbitmq-dotnet-client/api/RabbitMQ.Client.ConnectionFactory.html#RabbitMQ_Client_ConnectionFactory_Ssl

https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/v5.2.0/projects/client/Unit/src/unit/TestSsl.cs#L93

Even with `amqps://` you still have to set...

```
cf.Ssl.Enabled = true;
```

**Reply by ZedZipDev** [reply]
It did not help. 
But I have solved the problem this way:
` connectionFactory.Ssl.Version = System.Security.Authentication.SslProtocols.Tls12;`

In this case it works fine!

**Reply by lukebakken** [reply]
Thank you for letting us know!
